shims/spark34/src/main/scala/org/apache/spark/sql/execution/FileSourceScanExecShim.scala

```diff
@@ -98,7 +105,9 @@ abstract class FileSourceScanExecShim(
 
   @transient override lazy val dynamicallySelectedPartitions: Array[PartitionDirectory] = {
     val dynamicPartitionFilters =
-      partitionFilters.filter(isDynamicPruningFilter)
+      partitionFilters
+        .filter(isDynamicPruningFilter)
+        .filterNot(isBloomFilterMightContain)
```

Comment:
Do bloom filters still take effect in Gluten if we do this?

Reply:
It doesn't task effect in this case, and partition filter will not re-evalute in following `FilterExec` or other operator.

Reply:
So the PR completely disables runtime bloom-filter in Gluten? Am I missing something?

Reply:
> completely

@zhztheplayer Only disable the case when bloomfilter is applied on hive table's partition column ( join key is hive table's partition column). 

In most case, bloom filter is applied on data column.